pcsx2/GameDatabase.cpp

```diff
@@ -14,63 +14,193 @@
+
+		// Validate game fixes, invalid ones will be dropped!
+		for (std::string fix : safeGetStringList(node, "gameFixes"))
+		{
+			bool fixValidated = false;
+			for (GamefixId id = GamefixId_FIRST; id < pxEnumEnd; ++id)
+			{
+				std::string validFix = wxString(EnumToString(id)).Append(L"Hack").ToStdString();
+				if (validFix == fix)
+				{
+					fixValidated = true;
+					break;
+				}
+			}
+			if (fixValidated)
+			{
+				gameEntry.gameFixes.push_back(fix);
+			}
+			else
+			{
+				Console.Error(fmt::format("[GameDB] Invalid gamefix: '{}', specified for serial: '{}'. Dropping!", fix, serial));
+			}
 		}
-	}
-	return false;
-}
 
-// Gets a string representation of the 'value' for the given key
-wxString Game_Data::getString(const wxString& key) const {
-	for (auto it = kList.begin(); it != kList.end(); ++it) {
-		if (it->CompareKey(key)) {
-			return it->value;
+		if (YAML::Node speedHacksNode = node["speedHacks"])
+		{
+			for (YAML::const_iterator entry = speedHacksNode.begin(); entry != speedHacksNode.end(); entry++)
+			{
+				// Validate speedhacks, invalid ones will be skipped!
+				std::string speedHack = entry->first.as<std::string>();
+
+				// NOTE - currently only 1 speedhack!
+				if (speedHack != "mvuFlagSpeedHack")
+				{
+					Console.Error(fmt::format("[GameDB] Invalid speedhack: '{}', specified for serial: '{}'. Dropping!", speedHack, serial));
+					continue;
+				}
+
+				gameEntry.speedHacks[speedHack] = entry->second.as<int>();
+			}
 		}
+
+		gameEntry.memcardFilters = safeGetStringList(node, "memcardFilters");
+
+		if (YAML::Node patches = node["patches"])
+		{
+			for (YAML::const_iterator it = patches.begin(); it != patches.end(); ++it)
+			{
+				const YAML::Node& node = *it;
+				std::string crc = safeGetString(node, "crc", "default");
+
+				if (gameEntry.patches.count(crc) == 1)
+				{
+					Console.Error(fmt::format("[GameDB] Patch with duplicate CRC: '{}' detected for serial: '{}'.  Skipping patch.", crc, serial));
+					continue;
+				}
+
+				GameDatabaseSchema::Patch patchCol;
+
+				patchCol.author = safeGetString(node, "author");
+				patchCol.patchLines = safeGetMultilineString(node, "content");
+				gameEntry.patches[crc] = patchCol;
+			}
+		}
+	}
+	catch (YAML::RepresentationException e)
+	{
+		Console.Error(fmt::format("[GameDB] Invalid GameDB syntax detected on serial: '{}'.  Error Details - {}", serial, e.msg));
+		gameEntry.isValid = false;
 	}
-	return wxString();
+	return gameEntry;
 }
 
-void Game_Data::writeString(const wxString& key, const wxString& value) {
-	for (auto it = kList.begin(); it != kList.end(); ++it) {
-		if (it->CompareKey(key)) {
-			if( value.IsEmpty() )
-				kList.erase(it);
-			else
-				it->value = value;
-			return;
+GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::findGame(const std::string serial)
+{
+	if (gameDb.count(serial) == 1)
```

Comment:
```suggestion
	if (GameDB.count(serial) == 1)
```

Reply:
I see gameDB, GameDB and gameDb scattered around the files. Not sure if that's intentional.

Reply:
In this case `gameDb` is needed, as that's the variable's name.  Seems like the instances of `GameDB` are for logs / comments.